src/pathfora.js

```diff
@@ -2256,6 +2255,118 @@
 
   /**
    * @class
+   * @name Inline
+   * @description Inline Personalization
+   */
+  Inline = function () {
+    this.elements = [];
+    this.preppedElements = [];
+
+    /*
+     * @description Prepare all the elements with the attribute provided
+     * @param {attr} name of attribute to select by
+     */
+    this.prepElements = function (attr) {
+      var dataElements = {};
+
+      if (this.elements.length === 0) {
+        this.elements = document.querySelectorAll('[' + attr + ']');
+      }
+
+      var elements = this.elements;
+
+      for (var i = 0; i < elements.length; i++) {
+        if (elements[i].getAttribute(attr) !== null) {
+          var theElement = elements[i],
+              group = theElement.getAttribute('data-liogroup'),
+              type = theElement.getAttribute('data-liotype'),
+              trigger = theElement.getAttribute('data-liotrigger');
+
+          if (!group) {
+            group = 'default';
+          }
+
+          if (!dataElements[group]) {
+            dataElements[group] = [];
+          }
+
+          dataElements[group].push({
+            elem: theElement,
+            displayType: theElement.style.display,
+            group: group,
+            type: type,
+            trigger: trigger
+          });
+        }
+      }
+
+      return dataElements;
+    };
+
+    /*
+     * @description Show/hide the elements based on membership
+     */
+    this.procElements = function () {
+      var elementObj = this.prepElements('data-liotrigger');
```

Comment:
wondering if we should skip everything that follows if `elementObj === {}` so we dont alter the DOM when unnecessary. Seems like we would do a ton of processing / adding CSS even if there were no instances of `data-liotrigger`.